demo/api/src/products/entities/product.entity.ts

```diff
@@ -155,14 +156,24 @@ export class Product extends BaseEntity<Product, "id"> implements DocumentInterf
     @Field(() => ProductStatistics, { nullable: true })
     statistics?: Ref<ProductStatistics> = undefined;
 
-    @OneToMany(() => ProductVariant, (variant) => variant.product, { orphanRemoval: true })
+    @OneToMany(() => ProductColor, (variant) => variant.product, { orphanRemoval: true })
     @CrudField({
         resolveField: true, //default is true
         //search: true, //not yet implemented
         //filter: true, //not yet implemented
         //sort: true, //not yet implemented
         input: true, //default is true
     })
+    colors = new Collection<ProductColor>(this);
```

Comment:
I converted the existing demo product variants into product colors to serve as nested input example to use product variants as good main Property example